# Tugas
Nama: Ahmad Hamsa Pattuneri

NIM: H071201063

# Latihan 1
Silahkan Download sebuah dataset menggunakan API Kaggle

In [11]:
!kaggle datasets list -s Iris

ref                                                        title                                              size  lastUpdated          downloadCount  voteCount  usabilityRating  
---------------------------------------------------------  ------------------------------------------------  -----  -------------------  -------------  ---------  ---------------  
uciml/iris                                                 Iris Species                                        4KB  2016-09-27 07:38:05         310904       3067  0.7941176        
arshid/iris-flower-dataset                                 Iris Flower Dataset                               1010B  2018-03-22 15:18:06          65138        498  0.8235294        
vikrishnan/iris-dataset                                    Iris Dataset                                       999B  2017-08-03 16:00:44           4907         36  0.7647059        
chuckyin/iris-datasets                                     Iris datasets                       

In [10]:
!kaggle datasets download uciml/iris --unzip 


  0%|          | 0.00/3.60k [00:00<?, ?B/s]
100%|██████████| 3.60k/3.60k [00:00<00:00, 3.70MB/s]


# Latihan 2
Lakukan import Library Pandas dan Library Numpy

In [3]:
!pip install pandas
!pip install numpy

In [4]:
import pandas as pd
import numpy as np

# Latihan 3
Panggil file (load dataset) dengan format .csv untuk dataset mengenai bunga Iris yang sudah peserta unduh dari Kaggle, dan akan disimpan di dalam dataframe df. Lalu tampilkan 5 baris awal dataset dengan function head()

In [5]:
df1 = pd.read_csv('/Users/ahmad/Documents/iris.csv')
df1.head(5)

Id  SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm      Species
0   1            5.1           3.5            1.4           0.2  Iris-setosa
1   2            4.9           3.0            1.4           0.2  Iris-setosa
2   3            4.7           3.2            1.3           0.2  Iris-setosa
3   4            4.6           3.1            1.5           0.2  Iris-setosa
4   5            5.0           3.6            1.4           0.2  Iris-setosa

# Latihan 4
Tampilkan tipe data dari kolom yang ada pada dataset

In [6]:
df1.dtypes

Id                 int64
SepalLengthCm    float64
SepalWidthCm     float64
PetalLengthCm    float64
PetalWidthCm     float64
Species           object
dtype: object

# Latihan 5
Apakah tipe Data dari kolom berikut ini: (silakan diisi pada cell di bawah ini)

Kolom  "Id" memiliki tipe data =  int64
    
Kolom "SepalLengthCm" memiliki tipe data =  float64
                                                    
Kolom "Species" memiliki tipe data =  object

# Latihan 6
Hitunglah ukuran (jumlah baris dan kolom) dari dataset. Dengan menggunakan method function

In [7]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Id             150 non-null    int64  
 1   SepalLengthCm  150 non-null    float64
 2   SepalWidthCm   150 non-null    float64
 3   PetalLengthCm  150 non-null    float64
 4   PetalWidthCm   150 non-null    float64
 5   Species        150 non-null    object 
dtypes: float64(4), int64(1), object(1)
memory usage: 7.2+ KB


# Latihan 7
Berapakah jumlah baris, dan jumlah kolom pada dataset? (silakan diisi pada cell di bawah ini)

Jumlah Baris pada dataset adalah =  150

Jumlah kolom pada dataset adalah =  6

# Latihan 8
Tampilkan data yang hanya berisi kolom "Id" dan kolom "Species" dalam bentuk dataframe.

In [26]:
df1[["Id", "Species"]]

Id         Species
0      1     Iris-setosa
1      2     Iris-setosa
2      3     Iris-setosa
3      4     Iris-setosa
4      5     Iris-setosa
..   ...             ...
145  146  Iris-virginica
146  147  Iris-virginica
147  148  Iris-virginica
148  149  Iris-virginica
149  150  Iris-virginica

[150 rows x 2 columns]

# Latihan 9
Tampilkan data dengan dataframe, dan data yang ditampilkan adalah data pada baris dengan indeks 0 (nol) sampai dengan indeks 9 (sembilan)

In [20]:
df1[0:10]

Id  SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm      Species
0   1            5.1           3.5            1.4           0.2  Iris-setosa
1   2            4.9           3.0            1.4           0.2  Iris-setosa
2   3            4.7           3.2            1.3           0.2  Iris-setosa
3   4            4.6           3.1            1.5           0.2  Iris-setosa
4   5            5.0           3.6            1.4           0.2  Iris-setosa
5   6            5.4           3.9            1.7           0.4  Iris-setosa
6   7            4.6           3.4            1.4           0.3  Iris-setosa
7   8            5.0           3.4            1.5           0.2  Iris-setosa
8   9            4.4           2.9            1.4           0.2  Iris-setosa
9  10            4.9           3.1            1.5           0.1  Iris-setosa

# Latihan 10
Tampilkan data hanya kolom "Id" dan kolom "Species" dengan dataframe, dan yang ditampilkan adalah data pada baris dengan indeks 11 (sebelas) sampai dengan indeks 15 (limabelas)

In [25]:
df9 = df1[["Id", "Species"]]
df9[11:15]

Id      Species
11  12  Iris-setosa
12  13  Iris-setosa
13  14  Iris-setosa
14  15  Iris-setosa

# Latihan 11
Pada DataFrame dapat menampilkan beberapa baris pertama/terakhir dari dataset yang di load. Gunakan Method head() dan tail().

Latihan: Tampilkan data pada 8 (delapan) baris pertama dari dataset, dengan dataframe.

In [27]:
df1.head(8)

Id  SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm      Species
0   1            5.1           3.5            1.4           0.2  Iris-setosa
1   2            4.9           3.0            1.4           0.2  Iris-setosa
2   3            4.7           3.2            1.3           0.2  Iris-setosa
3   4            4.6           3.1            1.5           0.2  Iris-setosa
4   5            5.0           3.6            1.4           0.2  Iris-setosa
5   6            5.4           3.9            1.7           0.4  Iris-setosa
6   7            4.6           3.4            1.4           0.3  Iris-setosa
7   8            5.0           3.4            1.5           0.2  Iris-setosa

# Latihan 12
Tampilkan data pada 3 (tiga) baris terakhir dari dataset, dengan dataframe.

In [28]:
df1.tail(3)

Id  SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm  \
147  148            6.5           3.0            5.2           2.0   
148  149            6.2           3.4            5.4           2.3   
149  150            5.9           3.0            5.1           1.8   

            Species  
147  Iris-virginica  
148  Iris-virginica  
149  Iris-virginica

# Latihan 13
Hitung korelasi dari dataset. Dengan menggunakan method function

In [29]:
df1.corr()

Id  SepalLengthCm  SepalWidthCm  PetalLengthCm  \
Id             1.000000       0.716676     -0.397729       0.882747   
SepalLengthCm  0.716676       1.000000     -0.109369       0.871754   
SepalWidthCm  -0.397729      -0.109369      1.000000      -0.420516   
PetalLengthCm  0.882747       0.871754     -0.420516       1.000000   
PetalWidthCm   0.899759       0.817954     -0.356544       0.962757   

               PetalWidthCm  
Id                 0.899759  
SepalLengthCm      0.817954  
SepalWidthCm      -0.356544  
PetalLengthCm      0.962757  
PetalWidthCm       1.000000

# Latihan 14
Berdasarkan pada perhitungan korelasi di Latihan (11), apakah yang dapat Bapak/Ibu simpulkan sementara? Silakan tuliskan simpulan sementara Bapak/Ibu pada cell di bawah ini.

# Latihan 15
Hitung korelasi untuk kolom berikut ini: PetalLengthCm, PetalWidthCm

In [33]:
df15 = df1[["PetalLengthCm", "PetalWidthCm"]]
df15.corr()

PetalLengthCm  PetalWidthCm
PetalLengthCm       1.000000      0.962757
PetalWidthCm        0.962757      1.000000

# Latihan 16
Method "describe" secara otomatis melakukan komputasi statistik untuk semua continous variable. Secara default "describe" melakukan ignore terhadap variabel bertype objek.

Komputasi statistik yang dilakukan terdiri dari: count, mean, std, min, max, 25%, 75%, max.

Latihan: Gunakan method describe pada dataset yang sudah di load untuk semua continous variabel. (Dataset Iris.csv)

In [42]:
df1.describe(include = ['float', 'int'])

Id  SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm
count  150.000000     150.000000    150.000000     150.000000    150.000000
mean    75.500000       5.843333      3.054000       3.758667      1.198667
std     43.445368       0.828066      0.433594       1.764420      0.763161
min      1.000000       4.300000      2.000000       1.000000      0.100000
25%     38.250000       5.100000      2.800000       1.600000      0.300000
50%     75.500000       5.800000      3.000000       4.350000      1.300000
75%    112.750000       6.400000      3.300000       5.100000      1.800000
max    150.000000       7.900000      4.400000       6.900000      2.500000

# Latihan 17
Gunakan method describe pada dataset yang sudah di load untuk data bertype objek. (Dataset Iris.csv)

In [37]:
df1.describe(include = 'object')

Species
count           150
unique            3
top     Iris-setosa
freq             50

# Latihan 18
Gunakan method describe pada dataset yang sudah di load untuk semua type data (continous variabel dan type object).

In [45]:
df1.describe(include = ['float', 'int', 'object'])

Id  SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm  \
count   150.000000     150.000000    150.000000     150.000000    150.000000   
unique         NaN            NaN           NaN            NaN           NaN   
top            NaN            NaN           NaN            NaN           NaN   
freq           NaN            NaN           NaN            NaN           NaN   
mean     75.500000       5.843333      3.054000       3.758667      1.198667   
std      43.445368       0.828066      0.433594       1.764420      0.763161   
min       1.000000       4.300000      2.000000       1.000000      0.100000   
25%      38.250000       5.100000      2.800000       1.600000      0.300000   
50%      75.500000       5.800000      3.000000       4.350000      1.300000   
75%     112.750000       6.400000      3.300000       5.100000      1.800000   
max     150.000000       7.900000      4.400000       6.900000      2.500000   

            Species  
count           150  
unique            3  
top     Iris-setosa  
freq             50  
mean            NaN  
std             NaN  
min             NaN  
25%             NaN  
50%             NaN  
75%             NaN  
max             NaN

# Latihan 19
Hitunglah nilai mean dari dataset.

In [49]:
df1.mean()

C:\Users\ahmad\AppData\Local\Temp/ipykernel_4144/2053335143.py:1: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.
  df1.mean()


Id               75.500000
SepalLengthCm     5.843333
SepalWidthCm      3.054000
PetalLengthCm     3.758667
PetalWidthCm      1.198667
dtype: float64

# Latihan 20
Hitung nilai mean dari dataset untuk kolom PetalLengthCm.

In [50]:
df1["PetalLengthCm"].mean()

3.7586666666666693

# Latihan 21
Carilah nilai minimal dari dataset untuk kolom SepalWidthCm.

In [51]:
df1["SepalWidthCm"].min()

2.0

# Latihan 22
Hitunglah nilai mean dari dataset untuk kolom SepalLengthCm per Species dengan menggunakan metode groupby.

In [59]:
df1.groupby('Species').mean()[['SepalWidthCm']]

SepalWidthCm
Species                      
Iris-setosa             3.418
Iris-versicolor         2.770
Iris-virginica          2.974

# Latihan 23
Hitunglah frekuensi pada kolom 'Species' dengan menggunakan metode value_counts().

In [60]:
df1.value_counts('Species')

Species
Iris-setosa        50
Iris-versicolor    50
Iris-virginica     50
dtype: int64

# Latihan 24
Tampilkan perhitungan frekuensi pada kolom 'Species' dengan menggunakan metode value_counts() dalam bentuk dataframe.

In [76]:
df1["Species"].value_counts().to_frame()

Species
Iris-setosa           50
Iris-versicolor       50
Iris-virginica        50

# Latihan 25
Hitunglah frekuensi pada kolom 'PetalLenghCm' dengan menggunakan metode value_counts() dan dalam bentuk dataframe.

In [82]:
df1["PetalLengthCm"].value_counts().to_frame()

PetalLengthCm
1.5             14
1.4             12
5.1              8
4.5              8
1.6              7
1.3              7
5.6              6
4.7              5
4.9              5
4.0              5
4.2              4
5.0              4
4.4              4
4.8              4
1.7              4
3.9              3
4.6              3
5.7              3
4.1              3
5.5              3
6.1              3
5.8              3
3.3              2
5.4              2
6.7              2
5.3              2
5.9              2
6.0              2
1.2              2
4.3              2
1.9              2
3.5              2
5.2              2
3.0              1
1.1              1
3.7              1
3.8              1
6.6              1
6.3              1
1.0              1
6.9              1
3.6              1
6.4              1